Makefile

```diff
@@ -50,25 +55,22 @@ test:
 clean:
 	@rm -rf Binaries
 
-toolset: /tmp/$(ROSLYN_TOOLSET_NAME).tar.bz2  /tmp/$(MONO_TOOLSET_NAME).tar.bz2 /tmp/$(NUGET_PACKAGE_NAME).zip
-
 clean_toolset:
-	rm /tmp/$(ROSLYN_TOOLSET_NAME).tar.bz2 ; \
-	rm /tmp/$(MONO_TOOLSET_NAME).tar.bz2 ; \
-	rm /tmp/$(NUGET_PACKAGE_NAME).zip
-
-/tmp/$(ROSLYN_TOOLSET_NAME).tar.bz2:
-	@pushd /tmp/ ; \
-	curl -O https://dotnetci.blob.core.windows.net/roslyn/$(ROSLYN_TOOLSET_NAME).tar.bz2 ; \
-	tar -jxf $(ROSLYN_TOOLSET_NAME).tar.bz2
-
-/tmp/$(MONO_TOOLSET_NAME).tar.bz2:
-	@pushd /tmp/ ; \
-	curl -O https://dotnetci.blob.core.windows.net/roslyn/$(MONO_TOOLSET_NAME).tar.bz2 ; \
-	tar -jxf $(MONO_TOOLSET_NAME).tar.bz2
-
-/tmp/$(NUGET_PACKAGE_NAME).zip:
-	@pushd /tmp/ ; \
-	curl -O https://dotnetci.blob.core.windows.net/roslyn/$(NUGET_PACKAGE_NAME).zip ; \
-	unzip -uoq $(NUGET_PACKAGE_NAME).zip -d ~/ ; \
-	chmod +x ~/.nuget/packages/Microsoft.Build.Mono.Debug/14.1.0/lib/MSBuild.exe
+	@rm -rf $(TOOLSET_TMP_PATH)
+
+toolset: $(TOOLSET_TMP_PATH)/$(ROSLYN_TOOLSET_NAME) $(TOOLSET_TMP_PATH)/$(NUGET_PACKAGE_NAME).zip
+
+$(TOOLSET_TMP_PATH)/$(ROSLYN_TOOLSET_NAME): | $(TOOLSET_TMP_PATH)
```

Comment:
This is a `make` order-only dependency: https://www.gnu.org/software/make/manual/html_node/Prerequisite-Types.html